### ISSUE TITLE
Multi Variant time series data - not telemetry

### ISSUE THREAD

**Issue by Pooja199** [khundman/telemanom#39]
Hi. I am trying to use this approach for multi variant time series data present to find anomalies in payment gateway attributes like timeouts, latency, etc.. In the paper it is mentioned that I should be using the first dimension of each channel to be telemetry values.  Would it matter if I don't have any telemetry data to work with?

**Comment by Pooja199** [comment]
Also, are the channels like different time series data - power is a time series , radiation is a time series? 
If that's the case, in results.csv the anomalies are detected for each channel - so is it detecting anomalies in power , radiation etc? 
My use case is that i should be able to detect anomalies at a particular time-step considering all the features at once

**Comment by khundman** [comment]
Your data doesn't need to be telemetry data, the comment you are referring to means only that the dimension you are trying to predict (i.e. identify anomalies for) should be positioned as the first dimension. 

I'm not sure I understand your second question. Each telemetry channel gets its own model that's objective to predict future telemetry points for that channel. In addition to each channel's historical telemetry that is passed in as the first dimension of the input, a common set of encoded spacecraft command information is passed with that channel/model, making the input multivariate.

**Comment by Pooja199** [comment]
Okay that makes sense. 
I am training the model for 9000 points and its taking about 30 minutes for 1 epoch. The error is converging really soon (~2) even for a much lower learning rate - i tried changing the lr to 0.00001 . Why would this happen?

**Comment by Purav-Zumkhawala** [comment]
So for example let's take a data that has 3 values, value_0, value_1 and value_2 for a single timestamp and ground truth value. Now if I want to detect anomaly using this package the input would be 3 .npy files with each of the above values set as 1st dimension in 3 different files. Correct ? then if any one of the channel says that the given timestamp and value pair is anomalous it should be considered as anomaly ? My goal is to predict outlier points as correctly as possible to the ground truth values.

**Comment by khundman** [comment]
@Pooja199 it's impossible for me to help or make suggestions without understanding your data. But if your residuals are flat it sounds like a setup problem and unrelated to any hyperparameters. I would double check your implementation - do you see gradually decreasing loss when training?

@Purav-Zumkhawala no if you only care about anomalies in your ground truth channel then you only need a single `.npy` that is multidimensional with the ground truth at dimension 0 and values 0,1, and 2 in subsequent dimensions. In this sense you will use historical ground truth values as well as the other inputs to try and prediction future ground truth values. You only need one model per channel being evaluated. In our experiment setup we were interested in anomalies in any channel, hence having separate models/inputs for each.
